docs/authentication.md

```diff
@@ -0,0 +1,194 @@
+
+# Authenticating with the CE-API  
+  
+All read operations in the Contributor Environment API are publicly accessible. However, in order to create, update, or delete nodes, the request **MUST BE** authenticated with a JWT token.
+
+## Request a JWT token
+
+The Contributor Environment API has an endpoint to create a JWT token. This endpoint **MUST ONLY** be used in a secured environment since it uses a private API key which shouldn't be visible to users. Best practice for web applications is to request the JWT token in a backend environment and to only use the JWT token in the frontend.
```

Comment:
Many demos are currently using netlify for hosting js-only applications. This means that I'm not sure if they'll have control over a backend that lets them do this action without exposing the api key. Any suggestions on how they could do this? Maybe @asarasua has some thoughts here?

Reply:
Yes, that will be an issue indeed. In Netlify, functions (Lambda) can be used to serve a token. Ideally, that should be restricted with some user authentication. E.g. login with your credentials to get a token.

Reply:
Is your thought that each _user_ of the demos will get a separate token, or will the whole demo app have a single token?

Reply:
The demo app will have a single token, but it should only be handed to users if they log in. With this, each partner can decide how to implement this.

Thinking ahead, perhaps the above could become quite cumbersome. Would maybe a generic SSO be feasible for the TROMPA project?

Reply:
I believe that both WebIDs as used by Solid, and Neo4j, support authentication using OpenID Connect. Is that accurate? If so, would it make sense to incorporate this for this purpose? (i.e. users log in to a TROMPA app with their Web ID, using their vcard URI as described in #103, in order to obtain the required app token?)

Reply:
Ok, I will create a new issue for this topic.